### PR TITLE
Upgrade spark to 3.4.0 version.

### PR DESCRIPTION
**Pull request by milos-colic**
Upgrade delta to match spark.
Propagate the changes.

### CODE REVIEW

**Review by mjohns-databricks — APPROVED** [review]
ltgm

File: src/test/scala/com/databricks/labs/mosaic/models/knn/SpatialKNNTest.scala [hunk 1]
```diff
@@ -1,11 +1,40 @@
...
+import org.scalatest.flatspec.AnyFlatSpec
 
-class SpatialKNNTest extends MosaicSpatialQueryTest with SharedSparkSession with SpatialKNNBehaviors {
+class SpatialKNNTest extends AnyFlatSpec with SpatialKNNBehaviors with SparkSuite {
+
+    "Mosaic" should "run SpatialKNN without approximation" in {
+        var conf = new SparkConf(false)
+            .set(MOSAIC_INDEX_SYSTEM, "H3")
+            .set(MOSAIC_GEOMETRY_API, "JTS")
+            .set(MOSAIC_RASTER_API, "GDAL")
```
Comment (sllynn) [inline src/test/scala/com/databricks/labs/mosaic/models/knn/SpatialKNNTest.scala:19]: This is redundant now, isn't it?

**Review by sllynn — APPROVED** [review]
Looks good 👍

**Comment by sllynn** [comment]
Probably need to bump the spark version in the workflow matrix?

**Pull request merged**